src/main/webapp/scripts/edit_card.js

```diff
@@ -13,82 +13,83 @@
 // limitations under the License.
 
 /**
- * Query salecard from servlet and insert values in the form
- * TODO: fetch from servlet
- * TODO: handle case where a vendor does not have a card registered
+ * Query salecard from servlet
  */
-const querySalecard = () => {
+async function querySalecard() {
+  let vendor;
+  await fetch('/get-vendor', {method: 'POST'})
+      .then(response => response.json())
+      .then(vendorEntity => {
+        vendor = vendorEntity;
+      });
```

Comment:
What if the request goes wrong and the response sends an error?

Reply:
+1, we can handle this by using `response.status` to prove if the servlet send an error or not